src/unix/notbsd/linux/other/mod.rs

```diff
@@ -232,6 +258,12 @@ s_no_extra_traits! {
     }
 }
 
+impl siginfo_t {
+    pub unsafe fn si_addr(&self) -> *mut ::c_void {
+        self.sifields.sigfault.addr
+    }
+}
```

Comment:
IIRC the C macros work like methods right? E.g. `#define si_addr sifields.__sigfault__.addr` or something like that ? 

We usually add C macros as normal functions, but I think in this case it makes sense to add them as methods, since that is how they would be used in C.

Reply:
Yup, exactly.

Reply:
cc @alexcrichton I think this is the first time we add methods to `libc` structs, but that seems justified here.

Reply:
It seems reasonable to add methods here to me, but especially due to types like `siginfo_t` showing up on multiple platforms there will probably be a slow and steady trickle of issues to add the methods to all other platforms as well. I think that's fine though!

For guidelines I'd recommend that methods should probably be as simple as possible when they can be, ideally only for this use case of C has some macro that accesses a bunch of sub-fields which doesn't translate well to Rust. I wouldn't want to get more expansive to other more complicated computations.